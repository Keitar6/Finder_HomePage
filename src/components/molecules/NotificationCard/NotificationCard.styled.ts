import styled from 'styled-components';
import initialTheme from '../../../theme/initialTheme';

const typography = {
  ...initialTheme.typography.smallParagraph,
};
export const NotificationCardContainer = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  border: 1px solid transparent;
  gap: 1.5rem;
  padding: 0.75rem 2.5rem 0.75rem 1.25rem;
  border-bottom-color: ${initialTheme.palette.inputGrey};
  min-width: 400px;
  background-color: transparent;

  cursor: pointer;

  :hover {
    background-color: ${initialTheme.palette.notificationGreyHover};
  }
`;

export const Texts = styled.div`
  display: flex;
  justify-content: center;
  align-items: baseline;
  flex-direction: column;
  gap: 0.3rem;
  max-width: 255px;
`;
export const RemoveButtonWrapper = styled.div`
  position: absolute;
  right: 1rem;
`;
export const IconWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 20px;
`;
export const Dummy = styled.div`
  width: 20px;
`;
export const UndoButton = styled.div`
  color: ${initialTheme.palette.blueDark};
  text-decoration: underline;
  cursor: default;
  ${typography}
`;
