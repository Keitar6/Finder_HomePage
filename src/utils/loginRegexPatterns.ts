import { InputVariant, RegexPat } from '../components/atoms/Input/Input.types';
import { Getters } from '../types/shared.types';

type LoginRegexProps = Getters<InputVariant, RegexPat>;

const loginRegexPattern: LoginRegexProps = {
  search: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
};

export default loginRegexPattern;
