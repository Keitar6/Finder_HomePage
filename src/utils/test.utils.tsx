import { PropsWithChildren } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import ThemeProvider from '../theme/ThemeContext';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  initialUrl?: string;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    initialUrl = '/',
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  const Wrapper = ({ children }: PropsWithChildren<object>): JSX.Element => (
    <ThemeProvider>
        {children}
    </ThemeProvider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
