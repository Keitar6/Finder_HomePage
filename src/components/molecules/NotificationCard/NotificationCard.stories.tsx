import { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from '../../../theme/globalStyles';
import ThemeProvider from '../../../theme/ThemeContext';
import NotificationCard from './NotificationCard';
import { NotificationsMock } from '../../../contexts/GeneralContext';

const meta = {
  title: 'Molecules/NotificationCard',
  component: NotificationCard,
  tags: ['autodocs'],

  argTypes: {
    data: {
      description: 'Enter data',
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
  parameters: {
    backgrounds: { default: 'light' },
  },
} satisfies Meta<typeof NotificationCard>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: { data: NotificationsMock[0] },
};
