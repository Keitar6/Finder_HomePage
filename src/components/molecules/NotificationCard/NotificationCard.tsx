import Typography from '../../atoms/Typography/Typography';
import * as Styled from './NotificationCard.styled';
import Icon from '../../atoms/Icon/Icon';
import { useState } from 'react';
import { IconName } from '../../atoms/Icon/icon.types';

export type Notification = {
  title: string;
  description: string;
  iconName: IconName | null;
};
export type NotificationProps = {
  data: Notification;
};

const NotificationCard = ({ data: { title, description, iconName } }: NotificationProps) => {
  const [removeNotification, setRemoveNotification] = useState<boolean>(false);
  const [ifRemoveHovered, setIfRemoveHovered] = useState<boolean>(false);

  const removeIconHover = () => {
    setIfRemoveHovered(true);
  };
  const removeIconNoHover = () => {
    setIfRemoveHovered(false);
  };

  const deleteNotification = () => {
    setRemoveNotification(true);
  };

  const undoDeleteNotification = () => {
    setRemoveNotification(false);
  };

  return (
    <Styled.NotificationCardContainer>
      {!removeNotification ? (
        <>
          <Styled.IconWrapper>
            {iconName !== null ? (
              <Icon name={iconName} size={20} color='blueDark' />
            ) : (
              <Styled.Dummy />
            )}
          </Styled.IconWrapper>

          <Styled.Texts>
            <Typography variant='smallParagraph' style={{ fontWeight: 'bold', width: '100%' }}>
              {title}
            </Typography>
            <Typography variant='smallParagraph' color='inputGrey'>
              {description}
            </Typography>
          </Styled.Texts>
          <Styled.Dummy />
          <Styled.RemoveButtonWrapper
            onMouseOver={removeIconHover}
            onMouseOut={removeIconNoHover}
            onClick={deleteNotification}
          >
            <Icon name={'remove'} size={0} color={ifRemoveHovered ? 'text' : 'inputGrey'} />
          </Styled.RemoveButtonWrapper>
        </>
      ) : (
        <>
          <Styled.Dummy />

          <Styled.Texts>
            <Typography variant='smallParagraph' style={{ width: '100%' }}>
              This notification has been dismissed.
            </Typography>
            <Styled.UndoButton onClick={undoDeleteNotification}>Undo</Styled.UndoButton>
          </Styled.Texts>
          <Styled.Dummy />
        </>
      )}
    </Styled.NotificationCardContainer>
  );
};

export default NotificationCard;
