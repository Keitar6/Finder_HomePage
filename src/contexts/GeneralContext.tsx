import { createContext, PropsWithChildren, useState } from 'react';
import { Notification } from '../components/molecules/NotificationCard/NotificationCard';

export const NotificationsMock: Notification[] = [
  {
    title: 'Get 2 months of Disney Plus free',
    description: "Don't miss out! Get 3 months of streaming for the price of 1",
    iconName: null,
    isDissmised: false,
  },
  {
    title: 'Exclusive Finder offers ',
    description: "View deals that you can't find anywhere else",
    iconName: 'tag',
    isDissmised: false,
  },
];

type Value = {
  notifications: Notification[];
  setNotifications: React.Dispatch<React.SetStateAction<Notification[]>>;
};

const InitialValue: Value = {
  notifications: NotificationsMock,
  setNotifications: () => 1,
};

export const GeneralContext = createContext(InitialValue);

const GeneralProvider = ({ children }: PropsWithChildren) => {
  const [notifications, setNotifications] = useState<Notification[]>(NotificationsMock);

  const value: Value = {
    notifications,
    setNotifications,
  };

  return <GeneralContext.Provider value={value}>{children}</GeneralContext.Provider>;
};

export default GeneralProvider;
