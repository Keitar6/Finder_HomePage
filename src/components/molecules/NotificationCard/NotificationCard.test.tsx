import { cleanup } from '@testing-library/react';
import { renderWithProviders } from '../../../utils/test.utils';
import NotificationCard from './NotificationCard';
import { NotificationsMock } from '../../../contexts/GeneralContext';

describe('NotificationCard', () => {
  afterEach(cleanup);
  describe('NotificationCard component', () => {
    const NotificationCardRender: any = () =>
      renderWithProviders(<NotificationCard data={NotificationsMock[0]} />, {});
    it('should render correctly', () => {
      const NotificationCard = NotificationCardRender();
      expect(NotificationCard).toMatchSnapshot();
    });
  });
});
