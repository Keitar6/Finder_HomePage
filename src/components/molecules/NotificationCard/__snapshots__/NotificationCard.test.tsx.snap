// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationCard NotificationCard component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-aXZVg hFrbSd"
      >
        <div
          class="sc-fqkvVR dprRsV"
        >
          <div
            class="sc-dcJsrY cySWUE"
          />
        </div>
        <div
          class="sc-gEvEer xxCMM"
        >
          <p
            style="font-family: Roboto, Arial, sans-serif; font-weight: bold; font-size: 14px; color: rgb(36, 42, 47); width: 100%;"
          >
            Get 2 months of Disney Plus free
          </p>
          <p
            style="font-family: Roboto, Arial, sans-serif; font-weight: 500; font-size: 14px; color: rgb(169, 177, 183);"
          >
            Don't miss out! Get 3 months of streaming for the price of 1
          </p>
        </div>
        <div
          class="sc-dcJsrY cySWUE"
        />
        <div
          class="sc-eqUAAy fRlqEH"
        >
          <span />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-aXZVg hFrbSd"
    >
      <div
        class="sc-fqkvVR dprRsV"
      >
        <div
          class="sc-dcJsrY cySWUE"
        />
      </div>
      <div
        class="sc-gEvEer xxCMM"
      >
        <p
          style="font-family: Roboto, Arial, sans-serif; font-weight: bold; font-size: 14px; color: rgb(36, 42, 47); width: 100%;"
        >
          Get 2 months of Disney Plus free
        </p>
        <p
          style="font-family: Roboto, Arial, sans-serif; font-weight: 500; font-size: 14px; color: rgb(169, 177, 183);"
        >
          Don't miss out! Get 3 months of streaming for the price of 1
        </p>
      </div>
      <div
        class="sc-dcJsrY cySWUE"
      />
      <div
        class="sc-eqUAAy fRlqEH"
      >
        <span />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
